### PR TITLE
style(weave): chatview styling

### PR DESCRIPTION
**Pull request by jwlee64**
## Description
groups tool responses with messages
changes background colors
adds robot icon
removes horizontal rules

## Testing

How was this PR tested?

### CODE REVIEW

**Comment by adamwdraper** [comment]
I'm seeing styling issues on the beta link: https://www.loom.com/share/16a01ecb20394ea1b2f6b4e2d5d62144?sid=a0ed33d1-a9f5-4440-87be-768f52e7ea6f

**Review by tssweeney — COMMENTED** [review]
Probably good for Jamie to review

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ChatView/MessageList.tsx [hunk 1]
```diff
@@ -1,18 +1,78 @@
...
+          key={i}>
+          <MessagePanel index={m.original_index ?? i} message={m} />
+        </div>
       ))}
     </div>
   );
 };
+
+// Associates tool calls with their responses
+const processToolCallMessages = (messages: Messages) => {
```
Comment (jamie-rasmussen) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ChatView/MessageList.tsx:38]: Add return type annotation

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ChatView/MessagePanel.tsx [hunk 2]
```diff
@@ -7,34 +8,98 @@ import {ToolCalls} from './ToolCalls';
 import {Message} from './types';
 
 type MessagePanelProps = {
+  index: number;
   message: Message;
   isStructuredOutput?: boolean;
+  isChoice?: boolean;
```
Comment (jamie-rasmussen) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ChatView/MessagePanel.tsx:14]: This does not appear to be used

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ChatView/MessagePanel.tsx [hunk 3]
```diff
@@ -7,34 +8,98 @@ import {ToolCalls} from './ToolCalls';
...
+  isNested?: boolean;
+  pendingToolResponseId?: string;
```
Comment (jamie-rasmussen) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ChatView/MessagePanel.tsx:16]: This also appears to be unused

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ChatView/MessageList.tsx [hunk 4]
```diff
@@ -1,18 +1,78 @@
...
 
-export const MessageList = ({messages}: MessageListProps) => {
+export const MessageList = ({
+  messages,
+  scrollLastMessage = false,
+}: MessageListProps) => {
+  const lastMessageRef = useRef<HTMLDivElement>(null);
+  const processedMessages = processToolCallMessages(messages);
+
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (jwlee64) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ChatView/MessageList.tsx:18]: messagelist scrolls to the last message, chat view scrolls the last choice

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ChatView/MessagePanel.tsx (hunk 2 above)
Comment (jwlee64) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ChatView/MessagePanel.tsx:14]: i use this later for the message panel buttons

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ChatView/MessagePanel.tsx (hunk 3 above)
Comment (jwlee64) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ChatView/MessagePanel.tsx:16]: isNested is use for margins on the toolCallResponses below

isChoice
pnedingToolResponseId
are used for the messagePanel buttons i can remove for this pr

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ChatView/MessageList.tsx [hunk 5]
```diff
@@ -1,18 +1,78 @@
...
+  const processedMessages: Message[] = [];
+  for (let i = 0; i < messages.length; i++) {
+    const message = messages[i];
+
+    // If there are no tool calls, just add the message to the processed messages
+    // and continue to the next iteration.
+    if (!message.tool_calls) {
+      processedMessages.push({
+        ...message,
+        original_index: message.original_index ?? i,
```
[the remark this answers is not in the document]
Comment (jwlee64) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ChatView/MessageList.tsx:48]: ah for the messagePanelButtons

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ChatView/MessagePanel.tsx (hunk 2 above)
Comment (jwlee64) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/ChatView/MessagePanel.tsx:14]: removed

**Review by jamie-rasmussen — APPROVED** [review]

**Pull request merged**